controllers/add_services.go

```diff
@@ -66,26 +73,55 @@ func (a AddServices) Reconcile(r *FoundationDBClusterReconciler, context ctx.Con
 			}
 
 			serviceName, _ := internal.GetInstanceID(cluster, processGroup.ProcessClass, idNum)
-			existingService := &corev1.Service{}
-			err = r.Get(context, client.ObjectKey{Namespace: cluster.Namespace, Name: serviceName}, existingService)
+			service, err := internal.GetService(cluster, processGroup.ProcessClass, idNum)
 			if err != nil {
+				return &Requeue{Error: err}
+			}
 
-				if !k8serrors.IsNotFound(err) {
-					return &Requeue{Error: err}
-				}
-				service, err := internal.GetService(cluster, processGroup.ProcessClass, idNum)
+			existingService := &corev1.Service{}
+			err = r.Get(context, client.ObjectKey{Namespace: cluster.Namespace, Name: serviceName}, existingService)
+			if err == nil {
+				// Update the existing service
+				err = updateService(r, context, cluster, existingService, service)
 				if err != nil {
 					return &Requeue{Error: err}
 				}
-
+			} else if k8serrors.IsNotFound(err) {
+				// Create a new service
 				err = r.Create(context, service)
 
 				if err != nil {
 					return &Requeue{Error: err}
 				}
+			} else {
+				return &Requeue{Error: err}
 			}
 		}
 	}
 
 	return nil
 }
+
+// updateServices updates selected safe fields on a service based on a new
+// service definition.
+func updateService(r *FoundationDBClusterReconciler, context ctx.Context, cluster *fdbtypes.FoundationDBCluster, currentService *corev1.Service, newService *corev1.Service) error {
+	serviceLog := log.WithValues("namespace", cluster.Namespace, "cluster", cluster.Name, "service", currentService.Name)
+	originalSpec := currentService.Spec.DeepCopy()
+
+	currentService.Spec.Selector = newService.Spec.Selector
+
+	needsUpdate := !equality.Semantic.DeepEqual(currentService.Spec, *originalSpec)
```

Comment:
shouldn't currentService be `newService` ?

Reply:
We want to compare with the currentService here because the currentService is what we're passing to the Update call. We're only updating selected fields on the existing service, because Kubernetes sets fields behind the scenes and we don't want to overwrite that.